### PR TITLE
Closes #2022 - Add beneficiaries SBD payout

### LINKED ISSUE #2022: Community Request - Reward Beneficiaries should be paid based on payout setting
Currently if an author chooses 50/50 rewards for their post/comment, and they include beneficiaries, the beneficiaries do not receive their rewards as 50/50. This should be updated so that the beneficiaries payout is the same as the author’s.

### PR DESCRIPTION
**Pull request by TimCliff**
This PR closes issue #2022 by updating the beneficiaries payout to use the same payout setting as the author. For example, if the author selects the `50% / 50%` payout setting for the post, then all of the beneficiary payouts will be paid using the `50% / 50%` payout setting. This is a change from the current behavior which always pays beneficiaries using `100% SP`.

RPC nodes with the `legacy_comment_benefactor_reward_operation` or `comment_benefactor_reward_operation` options enabled will return data for those virtual operations in a different format when this change is applied. Instead of a single `reward` asset, there will now be three: `sbd_payout`, `steem_payout`, and `vesting_payout`.

### CODE REVIEW

File: libraries/chain/database.cpp [hunk 1]
```diff
@@ -1728,8 +1728,21 @@ share_type database::cashout_comment_helper( util::comment_reward_context& ctx,
             for( auto& b : comment.beneficiaries )
             {
                auto benefactor_tokens = ( author_tokens * b.weight ) / STEEM_100_PERCENT;
-               auto vest_created = create_vesting( get_account( b.account ), asset( benefactor_tokens, STEEM_SYMBOL ), has_hardfork( STEEM_HARDFORK_0_17__659 ) );
-               push_virtual_operation( comment_benefactor_reward_operation( b.account, comment.author, to_string( comment.permlink ), vest_created ) );
+
+               if( has_hardfork( STEEM_HARDFORK_0_20__2022 ) )
+               {
+                              auto benefactor_sbd_steem     = ( benefactor_tokens * comment.percent_steem_dollars ) / ( 2 * STEEM_100_PERCENT ) ;
```
Comment (mvandeberg) [inline libraries/chain/database.cpp]: What is with this indentation? Are you using tabs...?

File: libraries/protocol/hardfork.d/0_20.hf [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #define STEEM_HARDFORK_0_20__1877 (STEEM_HARDFORK_0_20)
 #define STEEM_HARDFORK_0_20__2019 (STEEM_HARDFORK_0_20)
 #define STEEM_HARDFORK_0_20__2428 (STEEM_HARDFORK_0_20)
+#define STEEM_HARDFORK_0_20__2022 (STEEM_HARDFORK_0_20)
```
Comment (mvandeberg) [inline libraries/protocol/hardfork.d/0_20.hf]: These should be in order, not appended.

**Comment by TimCliff** [comment]
updates made

**Comment by mvandeberg** [comment]
I've just realized, this unit test should be broken by these changes https://github.com/steemit/steem/blob/develop/tests/tests/operation_tests.cpp#L6703-L6727

And rightly so, as the PR changes how beneficiary payouts work. The branch `2555-PR` is now running through our CI. I will quickly find out if it is broken. Please check this and fix the tests and make the PR against the branch `2555-PR`.

**Comment by TimCliff** [comment]
looking into it now

**Comment by mvandeberg** [comment]
SBD print rate had some issues as well.

**Pull request merged**